awx/ui_next/src/components/DeleteButton/DeleteButton.jsx

```diff
@@ -12,48 +23,98 @@ function DeleteButton({
   variant,
   children,
   isDisabled,
+  deleteMessage,
+  deleteDetailsRequests,
 }) {
   const [isOpen, setIsOpen] = useState(false);
+  const [deleteMessageError, setDeleteMessageError] = useState();
+  const [deleteDetails, setDeleteDetails] = useState({});
 
+  const toggleModal = async isModalOpen => {
+    if (deleteDetailsRequests?.length && isModalOpen) {
+      const { results, error } = await getDeleteDetails(deleteDetailsRequests);
+      if (error) {
+        setDeleteMessageError(error);
+      } else {
+        setDeleteDetails(results);
+      }
+    }
+    setIsOpen(isModalOpen);
+  };
+  // console.log(deleteMessageError, 'here');
+
+  if (deleteMessageError) {
+    return (
+      <AlertModal
+        isOpen={deleteMessageError}
+        title={i18n._(t`Error!`)}
+        onClose={() => {
+          toggleModal(false);
+          setDeleteMessageError();
+        }}
+      >
+        <ErrorDetail error={deleteMessageError} />
+      </AlertModal>
+    );
+  }
   return (
     <>
       <Button
         variant={variant || 'secondary'}
         aria-label={i18n._(t`Delete`)}
         isDisabled={isDisabled}
-        onClick={() => setIsOpen(true)}
+        onClick={() => toggleModal(true)}
       >
         {children || i18n._(t`Delete`)}
       </Button>
-      <AlertModal
-        isOpen={isOpen}
-        title={modalTitle}
-        variant="danger"
-        onClose={() => setIsOpen(false)}
-        actions={[
-          <Button
-            key="delete"
-            variant="danger"
-            aria-label={i18n._(t`Delete`)}
-            isDisabled={isDisabled}
-            onClick={onConfirm}
-          >
-            {i18n._(t`Delete`)}
-          </Button>,
-          <Button
-            key="cancel"
-            variant="link"
-            aria-label={i18n._(t`Cancel`)}
-            onClick={() => setIsOpen(false)}
-          >
-            {i18n._(t`Cancel`)}
-          </Button>,
-        ]}
-      >
-        {i18n._(t`Are you sure you want to delete:`)}
-        <br />
-        <strong>{name}</strong>
-      </AlertModal>
+      {!deleteMessageError && (
```

Comment:
this check seems unnecessary since you've already returned on linke 47 if `deleteErrorMessage` is set